language/stdlib/modules/libra_system.mvir

```diff
@@ -68,16 +86,30 @@ module LibraSystem {
 
     // Set the metadata for the current block.
     // The runtime always runs this before executing the transactions in a block.
-    // TODO: make this private, support other metadata
-    public block_prologue(height: u64) acquires BlockMetadata, ValidatorSet {
+    // TODO: 1. Make this private, support other metadata
+    //       2. Should the previous block votes be provided from BlockMetadata or should it come from the ValidatorSet
+    //          Resource?
+    public block_prologue(
+        height: u64,
+        timestamp: u64,
+        new_block_hash: bytearray,
+        previous_block_votes: bytearray,
+        proposer: address
+    ) acquires BlockMetadata, ValidatorSet {
```

Comment:
can we separate the BlockMetadata function with sth like the Self.reconfigure? so that it's clear to read `block_prologue`
```
  Self.process_block_metadata();
  Self.reconfigure();
```